### ISSUE TITLE
Proposal: official tektoncd action(s)

### ISSUE THREAD

**Issue by vdemeester** [tektoncd/community#1113]
TL;DR The `tektoncd` community could provide _official_ action(s) to setup tektoncd components using GitHub workflows.

For our [Red Hat Tekton Catalog](https://github.com/openshift-pipelines/tektoncd-catalog) project (and other repositories as well), we use a mix of GitHub workflows and OpenShift based CI to tests our tasks. For this, we developed an [action to setup `tektoncd/pipeline`](https://github.com/openshift-pipelines/setup-tektoncd) in a job. Today it only installs `tektoncd/pipeline` and the cli, but we could definitely extend it to install other components (or have different actions).

This issue is to propose that the `tektoncd` community would maintain this action instead of `openshift-pipelines`. This would make it more visible and more **official** as it would be the same organization maintaining the tool and the action(s).

Note: it could be seen as an "help" somehow on adopting tekton as well 😛 

# Who will own it ?

- @vdemeester 
- @otaviof
- _[any other takers]_

/cc @jerop @dibyom @afrittoli @wlynch

**Comment by otaviof** [comment]
Thanks for bringing up this subject upstream, @vdemeester 🙏 Also, [shipwright-io](https://github.com/shipwright-io) uses an early version of the GhA, we should join efforts 👍

/cc @adambkaplan

**Comment by afrittoli** [comment]
This could be really handy, for testing purposes and for giving users a quick path to getting started with Tekton, so +1 from me. I might occasionally contribute but I would be able to maintain this though.

**Comment by jerop** [comment]
I support official actions to make it easier to integrate Tekton with GitHub Actions 

I'd put together a simple [GHA](https://github.com/jerop/tkn) to set up Tekton CLI, will deprecate it and ask the [users](https://github.com/search?q=jerop%2Ftkn&type=code) to rely on the official ones

**Comment by wlynch** [comment]
Generally sgtm! Only question I have is whether you'd want different actions to have different repos or have them all live in the same repo (mainly affects repo naming).

Single repo would make it a little easier to create new ones since we wouldn't have to raise a new issue to create a new repo for each action, though I'd recommend having a more general name (e.g. tektoncd/github-actions/setup-tektoncd)

**Comment by adambkaplan** [comment]
This would be excellent! On the Shipwright project, we have a mix of other actions and [make scripts](https://github.com/shipwright-io/build/blob/main/.github/workflows/ci.yml) to spin up tektoncd/pipeline on a KinD cluster.

**Comment by vdemeester** [comment]
Work-in-progress : https://github.com/tektoncd/actions 👼🏼

**Comment by vdemeester** [comment]
The repository exists, so I'll go ahead and close this 👼🏼
